packages/vite/src/node/shortcuts.ts

```diff
@@ -109,7 +109,11 @@ const BASE_SHORTCUTS: CLIShortcut[] = [
         return
       }
 
-      openBrowser(url, true, server.config.logger)
+      const path =
+        typeof server.config.server.open === 'string'
+          ? new URL(server.config.server.open, url).href
+          : ''
+      openBrowser(path ? path : url, true, server.config.logger)
```

Comment:
I think we should do a similar check to

https://github.com/vitejs/vite/blob/5bd6c0afd22d0ec7b99c23b491b2f8ce629e6272/packages/vite/src/node/server/index.ts#L677-L682

instead.

Reply:
I have also considered whether it is consistent here. I see `server.resolvedUrls?.local` has done similar processing. Perhaps we can directly deal with `server.resolvedUrls.local` in the following problems locally. Let the three act in unison

https://github.com/vitejs/vite/blob/8c87af7c03f5b989c8c86cf0c4efb0313c24af82/packages/vite/src/node/utils.ts#L870-L879

Reply:
Yeah it would be great too to have all of them use `server.resolvedUrls.local` when opening the browser. Feel free to refactor them if you'd like! I'm also fine to duplicate the code a little as the refactor can always be done separately.